src/modules/editor/SoundEditor/Note.ts

```diff
@@ -1,49 +1,52 @@
-
 import * as Tone from "tone";
 import { SampleLibrary } from "@modules/editor/SoundEditor/Tonejs-Instruments";
 
-class Note {
-  public samp: Tone.Sampler;
-  constructor(
-    public note: string = "Nan",
-    private _duration: number = 1,
-    public _instrument: string = "piano",
-  ) {
-    this.note = note;
-    if (this.note == "Nan") {
-      this.samp = new Tone.Sampler();
-    } else {
-      this.samp = SampleLibrary.load({
-        instruments: this._instrument,
-
-      }).toDestination();
-    }
-
-  }
-
-  public get duration() : number {
-    return this._duration;
-  }
+export interface NoteData {
+  note: string;
+  duration: number;
+  instrument: string;
+}
 
-  public toJson() : string {
-    return JSON.stringify({
-      note: this.note,
-      duration: this._duration,
-      instrument: this._instrument
-    });
+// Create a single sampler instance for each instrument
+const samplers = new Map<string, Tone.Sampler>();
+
+export const createNote = (
+  note: string = "Nan",
```

Comment:
I would prefer you to use a more appropriate value for a non-set value. You can do `note?: string` (or `note: ?string`, I never know the syntax for this stuff), and then check if this is `undefined` over a constant.